src/DependencyInjection/Admin/AdminConfiguration.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\DependencyInjection\Admin;
+
+use Doctrine\Common\Annotations\Annotation\NamedArgumentConstructor;
+
+
+/**
+ * @Annotation
+ * @NamedArgumentConstructor
+ * @Target({"CLASS"}) 
+ */
+#[\Attribute(\Attribute::TARGET_CLASS)]
```

Comment:
I guess it doesn't work for projects that use `php 7.4`

Reply:
Yes it does, using annotation. This class use both features annotation and attributes so it's compatible with php 7. It's the same logic used by symfony's contraints

Reply:
ohhh I got it, then for php 7.4 it'll be just a simple comment `#[\Attribute(\Attribute::TARGET_CLASS)]`

I got it!

Reply:
yes indeed !